pcsx2/DEV9/ATA/Commands/ATA_SCE.cpp

```diff
@@ -49,7 +59,33 @@ void ATA::SCE_IDENTIFY_DRIVE()
 {
 	PreCmd();
 
-	//HDD_IdentifyDevice(); //Maybe?
+	// fill sceSec response with default data
+	memcpy(sceSec, "Sony Computer Entertainment Inc.", 32);
+	memcpy(sceSec + 0x20, "SCPH-20401", 10);
+	memcpy(sceSec + 0x30, "  40", 4);
+
+	sceSec[0x40] = 0; // 4-byte HDD internal SCE serial, always starts with 0x02
+	sceSec[0x41] = 0;
+	sceSec[0x42] = 0;
+	sceSec[0x43] = 0x02;
+
+	sceSec[0x46] = 0x1a; // unknown 8-bytes
+	sceSec[0x47] = 0x01;
+	sceSec[0x48] = 0x02;
+	sceSec[0x49] = 0x20;
+	sceSec[0x4c] = 0x01;
+	sceSec[0x4d] = 0x03;
+	sceSec[0x4e] = 0x11;
+	sceSec[0x4f] = 0x01;
+	// 0x50 - 0x80 is a random unique block of data
+
+	// if exists .hddid file along with HDD image, use its content for sceSec response
+	std::string hddid_file(Path::ReplaceExtension(EmuConfig.DEV9.HddFile.c_str(), "hddid"));
+	auto fp = FileSystem::OpenManagedCFile(hddid_file.c_str(), "rb");
+	if (fp && FileSystem::FSize64(fp.get()) >= 128)
+		std::fread(sceSec, 1, 128, fp.get());
+	else
+		DevCon.WriteLn("Failed to handle HDD ID file '%s'", hddid_file.c_str());
 
 	pioDRQEndTransferFunc = nullptr;
 	DRQCmdPIODataToHost(sceSec, 256 * 2, 0, 256 * 2, true);
```

Comment:
Does the SCE Security Identify send a full 256 words or is it sending less data?

I note that the hddid file is expected to be 128 bytes (64 words).

Reply:
its sends full data (256 x 2), but the rest of response is zero filled in ATA.h

Reply:
So only first 128 bytes contains necessary information.

Reply:
I was more wondering if real hardware sends the full 256 x 2 bytes for this command

Reply:
Ah, okay